pallets/asset-manager/src/lib.rs

```diff
@@ -73,11 +74,15 @@ pub mod pallet {
 		fn get_asset_type(asset_id: T::AssetId) -> Option<T::AssetType> {
 			AssetIdType::<T>::get(asset_id)
 		}
+
+		fn get_asset_id(asset_type: T::AssetType) -> Option<T::AssetId> {
+			AssetTypeId::<T>::get(asset_type)
+		}
```

Comment:
This implies a 1:1 mapping between an `AssetId` and an `AssetType` (instead of a many-to-one). I previously thought it made sense to support multiple `AssetId`s mapping to a single `AssetType`, is this categorically wrong? Or is it not the `AssetManager`s responsibility to handle this?

Reply:
I think this is wrong. We need the asset-manager to provide one specific multiLocation when a user says: "I want to transact asset X".

Reply:
The assetType exists mainly to be able to reference distinct type of tokens in the future. Right now its an enum containing XCM(MultiLocation) but in the future we might want to distinguish between, e.g., fungible and non-fungible tokens, which should probably go into different pallets. But since our approach is to retrieve the Id through a hash, there is low risk of collision

Reply:
Oh, that's right. This constraint is about communicating back to another chain, right?

Reply:
yes